### DISCUSSION TITLE
broader std::function  support?

### DISCUSSION THREAD

**Discussion by ska-libang** [arkhipenko/TaskScheduler#173] [Ideas]
@arkhipenko   have you looked at broadening support for std::function  as Task?  It's currently limited to some particular ARDUINO defines, but preliminary testing suggests it will work fine with newer gcc toolchains for e.g. teensy 4.1 (arm M7)

Was there reasoning beyond "not tested" for erroring the _TASK_STD_FUNCTION  unless it's on the list of boards?

**Comment by arkhipenko** [comment]
@ska-libang  - No reasoning other than I am not using std::function that much (if at all), hence "not tested". 
If you have time, please do suggest how it could be expanded. 
Thank you!

**Reply by ska-libang** [reply]
Ok, I'll have a look. I had a codebase that was already using std::function objects as callback objects, so a bit of friction introducing your scheduler. 

Are you thinking about a release (3.8?) soon?

**Reply by ska-libang** [reply]
I guess one follow up.  Currently you let the user signal they want to use std::function<void()>  rather than void *()  by passing _TASK_STD_FUNCTION   define, but later error out if it doesn't match a particular set of boards.   

So in theory removeing the test in 285-290 TaskScheduler.h (current main) would achieve this with a warning about a untested board, this is what I did locally with tests.  But you may want to be more careful than that?

**Comment by mattbaker-digital** [comment]
I too would like OOTB support for Teensy if possible.  
~~Perhaps a _TASK_ALLOW_UNTESTED flag which changes it from an error to a warning without having to modify the library source.~~
#184 details a better method using __cplusplus and __has_cpp_attribute.

TaskScheduler.h and TaskSchedulerDeclarations.h also suggest that only the ESP8266 is supported when others have since been added.
